tasks/pkg-redhat.yml

```diff
@@ -23,32 +23,52 @@
     state: present
   when: not ansible_check_mode
 
-- name: Install Datadog yum repo
+- name: Install Datadog Agent 5 yum repo
   yum_repository:
-    name: "ansible_datadog_{{ item.key }}"
+    name: datadog
     description: Datadog, Inc.
-    baseurl: "{{ item.value }}"
+    baseurl: "{{ datadog_agent5_yum_repo }}"
     enabled: yes
     gpgcheck: yes
-    gpgkey: "{% if datadog_agent_major_version|int == 7 %}{{ datadog_yum_gpgkey_20200908 }}{% else %}{{ datadog_yum_gpgkey }}{% endif %}"
-    state: "{% if item.key == datadog_agent_major_version|int and datadog_yum_repo | length == 0 %}present{% else %}absent{% endif %}"
-  when: (not ansible_check_mode)
-  with_dict:
-    5: '{{ datadog_agent5_yum_repo }}'
-    6: '{{ datadog_agent6_yum_repo }}'
-    7: '{{ datadog_agent7_yum_repo }}'
-
-- name: (Custom) Install Datadog yum repo
+    gpgkey: "{{ datadog_yum_gpgkey }}"
+  when: (datadog_agent_major_version|int == 5) and (datadog_yum_repo | length == 0) and (not ansible_check_mode)
+
+- name: Install Datadog Agent 6 yum repo
+  yum_repository:
+    name: datadog
+    description: Datadog, Inc.
+    baseurl: "{{ datadog_agent6_yum_repo }}"
+    enabled: yes
+    gpgcheck: yes
+    gpgkey: "{{ datadog_yum_gpgkey }}"
+  when: (datadog_agent_major_version|int == 6) and (datadog_yum_repo | length == 0) and (not ansible_check_mode)
+
+- name: Install Datadog Agent 7 yum repo
   yum_repository:
-    name: ansible_datadog_custom
+    name: datadog
+    description: Datadog, Inc.
+    baseurl: "{{ datadog_agent7_yum_repo }}"
+    enabled: yes
+    gpgcheck: yes
+    gpgkey: "{{ datadog_yum_gpgkey_20200908 }}"
+  when: (datadog_agent_major_version|int == 7) and (datadog_yum_repo | length == 0) and (not ansible_check_mode)
+
+- name: Install Datadog Custom yum repo
+  yum_repository:
+    name: datadog
     description: Datadog, Inc.
     baseurl: "{{ datadog_yum_repo }}"
     enabled: yes
     gpgcheck: yes
     gpgkey: "{{ datadog_yum_gpgkey }}"
-    state: present
   when: (datadog_yum_repo | length > 0) and (not ansible_check_mode)
 
+- name: Remove old yum repo files
+  yum_repository:
+    name: "ansible_datadog_{{ item }}.repo"
+    state: absent
+  with_items: [ 5, 6, 7 ]
```

Comment:
I think we also want to run this with `custom`, to remove the custom yum repository (if it exists).

Reply:
Done and fixed the tests that were failing.